miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java

```diff
@@ -356,21 +354,29 @@ private static DetailedSample toDetailedSample(DetailedSampleDto from) {
   private static DetailedSample getParent(DetailedSampleDto childDto) {
     DetailedSample parent = null;
     if (childDto.getParentId() != null) {
-      // Case 1
       parent = new DetailedSampleImpl();
       parent.setId(childDto.getParentId());
-    } else if (childDto instanceof SampleIdentityDto && childDto.getClass() != SampleIdentityDto.class) {
-      // Case 2 or 3 require Identity details
-      Identity identity = toIdentitySample((SampleIdentityDto) childDto);
-      if (childDto instanceof SampleTissueDto && childDto.getParentSampleClassId() != null) {
-        // Case 2
-        parent = toTissueSample((SampleTissueDto) childDto);
-        parent.setSampleClass(new SampleClassImpl());
-        parent.getSampleClass().setId(childDto.getParentSampleClassId());
-        parent.setParent(identity);
-      } else {
-        // Case 3
-        parent = identity;
+    } else {
+      if (childDto instanceof SampleIdentityDto && childDto.getClass() != SampleIdentityDto.class) {
+        parent = toIdentitySample((SampleIdentityDto) childDto);
+      }
+
+      if (childDto instanceof SampleTissueDto && childDto.getClass() != SampleTissueDto.class) {
+        if (childDto.getParentSampleClassId() == null) {
```

Comment:
Please rename DetailedSample.parentSampleClassId to indicate that it refers to the tissue-level parent only, and not always the direct parent